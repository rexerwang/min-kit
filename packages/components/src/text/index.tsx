import { isNumber, isString } from '@min-kit/shared'
import { Text, TextProps } from '@tarojs/components'

export function renderText(text?: React.ReactNode, props: Omit<TextProps, 'children' | 'decode' | 'space'> = {}) {
  if (isString(text) || isNumber(text)) {
    return (
      <Text {...props} decode space='nbsp'>
        {text}
      </Text>
    )
  }

  return <>{text}</>
}
