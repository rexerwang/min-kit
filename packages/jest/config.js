const defineJestConfig = require('@tarojs/test-utils-react/dist/jest.js').default

module.exports = defineJestConfig({
  setupFilesAfterEnv: [require.resolve('./config/setupAfterEnv.js')],
  testEnvironment: 'jsdom',
  testMatch: ['<rootDir>/src/__tests__/**/*.test.ts?(x)'],
  globals: require('./config/globals'),
  collectCoverageFrom: ['<rootDir>/src/**/*.ts?(x)', '!**/__tests__/**', '!**/{debounce,throttle}.ts'],
  coverageReporters: ['clover', 'json', 'lcov', 'text-summary'],
  passWithNoTests: true,
})
