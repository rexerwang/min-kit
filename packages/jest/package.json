{
  "name": "@min-kit/jest",
  "version": "0.5.11",
  "private": true,
  "description": "jest config & utils for internal",
  "author": "rexerwang",
  "license": "MIT",
  "repository": "rexerwang/min-kit",
  "exports": {
    ".": "./utils.ts",
    "./config": "./config.js"
  },
  "devDependencies": {
    "@happy-dom/jest-environment": "^14.7.1",
    "@tarojs/test-utils-react": "^0.1.1",
    "@testing-library/jest-dom": "^6.4.8",
    "@testing-library/react": "^16.0.1"
  }
}
