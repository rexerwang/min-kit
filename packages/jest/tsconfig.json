{
  "extends": "../../tsconfig.root.json",
  "include": ["utils.ts"]
}
