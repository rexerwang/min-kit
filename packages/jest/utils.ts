import TestUtils from '@tarojs/test-utils-react'

export const createTaroTestUtils = () => {
  const testUtils = new TestUtils()
  return Object.assign(testUtils, { unmount: testUtils.unmout })
}

export const spyOnConsole = (methods = ['error', 'warn', 'info', 'debug']) =>
  methods.map((method: any) => {
    const spy = jest.spyOn(console, method)
    spy.mockImplementation(jest.fn())
    return spy
  })

export * from '@testing-library/react'
