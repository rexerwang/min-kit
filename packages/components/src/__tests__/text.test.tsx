import { render } from '@min-kit/jest'

import { MinText, renderText } from '../text'

describe('<MinText />', () => {
  it('should render <MinText /> toMatchSnapshot', async () => {
    const { asFragment } = render(<MinText className='test'>text</MinText>)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should renderText string toMatchSnapshot', () => {
    const Text = renderText('text', { userSelect: true })
    expect(render(Text).asFragment()).toMatchSnapshot()
  })

  it('should renderText JSX toMatchSnapshot', () => {
    const Text = renderText(<div>text</div>, { userSelect: true })
    expect(render(Text).asFragment()).toMatchSnapshot()
  })
})
