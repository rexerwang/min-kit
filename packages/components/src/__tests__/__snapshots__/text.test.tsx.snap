// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MinText /> should render <MinText /> toMatchSnapshot 1`] = `
<DocumentFragment>
  <taro-text-core
    class="test"
    decode="true"
    space="nbsp"
  >
    <!---->
    text
  </taro-text-core>
</DocumentFragment>
`;

exports[`<MinText /> should renderText JSX toMatchSnapshot 1`] = `
<DocumentFragment>
  <div>
    text
  </div>
</DocumentFragment>
`;

exports[`<MinText /> should renderText string toMatchSnapshot 1`] = `
<DocumentFragment>
  <taro-text-core
    class=""
    decode="true"
    space="nbsp"
    user-select="true"
  >
    <!---->
    text
  </taro-text-core>
</DocumentFragment>
`;
